src/auto/assistant.rs

```diff
@@ -233,9 +234,27 @@ impl<O: IsA<Assistant>> AssistantExt for O {
         }
     }
 
-    //fn set_forward_page_func<'a, P: Into<Option<&'a /*Unimplemented*/AssistantPageFunc>>, Q: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, page_func: P, data: Q, destroy: /*Unknown conversion*//*Unimplemented*/DestroyNotify) {
-    //    unsafe { TODO: call ffi::gtk_assistant_set_forward_page_func() }
-    //}
+    fn set_forward_page_func<P: Fn(i32) -> i32 + 'static, Q: Into<Option<P>>>(&self, page_func: Q) {
+        let page_func = page_func.into();
+        let page_func_data: Box_<Box_<Option<P>>> = Box::new(Box::new(page_func.into()));
+        unsafe extern "C" fn page_func_func<P: Fn(i32) -> i32 + 'static>(current_page: libc::c_int, data: glib_ffi::gpointer) -> libc::c_int {
+            let callback: Box_<Box_<Option<P>>> = Box_::from_raw(data as *mut _);
```

Comment:
`from_raw()` will cause the closure to be freed at the end of the scope. Use simple references here instead, or `into_raw()` again at the end but that's less nice.

```rust
let callback = &*(data as *const Box_<Option<P>>);
```